rust/otel-arrow-rust/src/otlp/attributes/decoder.rs

```diff
@@ -62,3 +76,532 @@ where
         }
     }
 }
+
+/// Decodes the parent IDs from their transport optimized encoding to the actual ID values.
+///
+/// In the transport optimized encoding, the record batch is sorted by value Type, then
+/// attribute key, then attribute value. If subsequent rows have the same key & value, the
+/// parent IDs are delta encoded.
+///
+/// There are additional exceptions that end the sequence of delta encoding:
+/// - null values (we don't consider null values equal, even if current row & previous
+///   row value are both null)
+/// - value types Empty, Slice & Map are never considered equal
+///
+/// For example:
+///
+/// | attr key |  attr val  | parent_id |
+/// |----------|------------| --------- |
+/// |   "a1"   |  str("a")  |  0        | <-- parent id = 0
+/// |   "a1"   |  str("a")  |  0        | <-- key & val == previous row -> delta encoded -> parent id = prev id + 0 = 0
+/// |   "a1"   |  str("a")  |  1        | <-- key & val == previous row -> delta encoded -> parent id = prev id + 1 = 1
+/// |   "a1"   |  str("a")  |  2        | <-- key & val == previous row -> delta encoded -> parent id = prev id + 2 = 3
```

Comment:
Not super familiar with arrow's delta encoding, but I was wondering if there should be a `2` here instead of `3`.
 
```suggestion
/// |   "a1"   |  str("a")  |  2        | <-- key & val == previous row -> delta encoded -> parent id = prev id + 2 = 2
```

Reply:
In this comment, I was trying to explain that the "prev id" was the ID from line 97, which is 1 and the delta is 2.

 1 (prev id) + 2 (delta) = 3